src/routing_table.rs

```diff
@@ -266,7 +266,7 @@ impl RoutingTable {
         if self.routing_table.len() < GROUP_SIZE {
             return true;
         }
-        let furthest_close_node = self.routing_table[GROUP_SIZE].clone();
+        let furthest_close_node = self.routing_table[GROUP_SIZE - 1].clone();
```

Comment:
Found a bug I introduced in RoutingTable before; will write accompanying test for `address_in_our_close_group`